apps/app.go

```diff
@@ -18,9 +18,52 @@ func (at AppType) IsValid() bool {
 		at == AppTypeBuiltin
 }
 
+// AssetType describes static assets of the Mattermost App.
+// Assets can be saved in S3 with appropriate permissions,
+// or they could be fetched as ordinary http resources.
+type AssetType string
+
+const (
+	s3Asset   AssetType = "s3_asset"
+	httpAsset AssetType = "http_asset"
+)
+
+func (at AssetType) IsValid() bool {
+	return at == s3Asset ||
+		at == httpAsset
+}
+
+// AppStatus describes status of the app
+type AppStatus string
+
+const (
+	AppStatusRegistered AppStatus = "registered"
+	AppStatusEnabled    AppStatus = "enabled"
```

Comment:
When do we set this flag on?

Reply:
It will be set by the system admin from the marketplace I think. We should provide the REST API to do it. There is a ticket for that.

Reply:
cc @hanzei

Reply:
@catalintomai After syncing up with Ben, we came to the conclusion that installed app is automatically enabled(as configuring the app is the part of the installation process).

Reply:
@iomodo, thanks, does that mean that we can remove that ticket you mentioned?